### ISSUE TITLE
(payments) Increase App.tsx/index.tsx test coverage, use AppErrorBoundary from shared React

### ISSUE THREAD

**Issue by LZoog** [mozilla/fxa#5338]
The `AppErrorBoundary` component in `fxa-components` (to become `fxa-react`) is currently identical to the one in payments and the tests are nearly identical. We can remove the declaration out of `App.tsx` and import the shared `AppErrorBoundary` instead.

When using the shared `AppErrorBoundary`, our test coverage in `App.tsx` doesn't quite meet the threshold. Our test threshold in `index.tsx` is pretty low as well, both files could be improved.

Run `CI=yes npm run test:frontend` to see the more detailed output.

Before sharing:
![image](https://user-images.githubusercontent.com/13018240/82078892-87300f00-96a7-11ea-8387-f6bb250ceac4.png)

After sharing:
![image](https://user-images.githubusercontent.com/13018240/82079170-06254780-96a8-11ea-9e73-bedf767085ce.png)

![image](https://user-images.githubusercontent.com/13018240/82079108-e8f07900-96a7-11ea-88db-5a76b2b7820b.png)

**Comment by lmorchard** [comment]
Yeah, index.tsx & App.tsx have some things in them that were particularly hard to mock & test when I went through coverage improvements last year. In particular, various APIs from browser, redux, stripe, and l10n that I tried to concentrate there and away from the rest of the app code.

Would be really nice to get 90%+ coverage there, but it seemed like a slog at the time. Might not be as bad as it felt on a second look

**Comment by lmorchard** [comment]
FWIW, though there's room for improvement on verifying props in mocked components, this PR technically improves App component coverage to 100%

https://github.com/mozilla/fxa/pull/6007
